src/features/import/utils/parseFile.ts

```diff
@@ -0,0 +1,130 @@
+import * as XLSX from 'xlsx';
+import { parse } from 'papaparse';
+
+export type ImportedFile = {
+  sheets: Sheet[];
+  title: string;
+};
+type Sheet = {
+  data: Row[];
+  title: string;
+};
+type Row = {
+  data: (string | number | null)[];
+};
+
+type ListMeta = {
+  items: {
+    data: (string | number | null)[];
+  };
+};
+
+export function createList(meta: ListMeta) {
+  return {
+    items: createListItems(meta.items.data),
+  };
+}
+
+export function createListItems(rawList: (string | number | null)[]) {
+  return rawList.map((i) => createListItem(i));
+}
+
+export function createListItem(data: string | number | null) {
+  return {
+    data,
+  };
+}
+
+export async function parseCSVFile(file: File): Promise<ImportedFile> {
+  return new Promise((resolve, reject) => {
+    const rawData: ImportedFile = {
+      sheets: [],
+      title: file.name,
+    };
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+
+    reader.onloadend = function () {
+      parse(file, {
+        complete: (result) => {
+          if (result.data) {
+            const sheetObject = {
+              data: result.data,
+              title: file.name,
+            };
+            rawData.sheets = [sheetObject as Sheet];
```

Comment:
Why is this type-cast (`as Sheet`) necessary? Is it because of the type of `result.data` not matching the type of `Sheet['data']`? If so, I would prefer the cast to be on line 52 instead, to more clearly indicate that that's the type mismatch.

But again, I'm suggesting @ziggabyte makes these changes as part of his work in the next step for this file.